src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -420,6 +421,7 @@ def get_objects_owned_by_application(self) -> List[ApplicationOwnedObject]:
             ).fetchall()
             return [{"name": row[1], "type": row[2]} for row in results]
 
+    @start_cli_metrics_span("create_or_upgrade_app")
```

Comment:
spans shouldn't be implementation-specific, but rather represent a logic step in the overall operation being performed.

Reply:
Adjusted the names and locations of all the spans to better encapsulate this, let me know if they work for you!